x/nexus/keeper/transfer.go

```diff
@@ -84,27 +135,42 @@ func (k Keeper) EnqueueForTransfer(ctx sdk.Context, sender exported.CrossChainAd
 		return 0, fmt.Errorf("recipient's chain %s does not support foreign assets", recipient.Chain.Name)
 	}
 
+	// merging incomplete transfers for the specified recipient
+	incompleteTransfer, found := k.getTransferForRecipientAndAsset(ctx, recipient, asset.Denom, exported.Incomplete)
+	if found {
+		asset = asset.Add(incompleteTransfer.Asset)
+		k.deleteTransfer(ctx, incompleteTransfer)
+	}
+
 	// collect fee
-	if feeDue := sdk.NewDecFromInt(asset.Amount).Mul(feeRate).TruncateInt(); feeDue.IsPositive() {
-		k.addTransferFee(ctx, sdk.NewCoin(asset.Denom, feeDue))
-		asset = asset.SubAmount(feeDue)
+	fees, _ := k.ComputeTransferFee(ctx, sender.Chain, recipient.Chain, asset)
+	if fees.Amount.GTE(asset.Amount) {
+		k.Logger(ctx).Debug(fmt.Sprintf("skipping deposit for chain %s at %s from recipient %s due to deposited amount being below "+
+			"fees %s for asset %s", sender.Chain.Name, sender.Address, recipient.Address, fees.String(), asset.String()))
+
+		return k.setNewTransfer(ctx, recipient, asset, exported.Incomplete), nil
+	}
+
+	if fees.IsPositive() {
```

Comment:
How can fee be <= here?

Reply:
If the `MinFee` itself is set to `0`, then it can be.

Reply:
how? this seems like it shouldn't be possible, i.e. something needs to change if it is

Reply:
If the `MinFee` and `FeeRate` for an asset on a chain are set to `0`, then transfer fees will be `0`. We do want to allow this, for e.g. incentives for transferring an asset, or if we need to update a token contract and do a swap, don't want to charge users for it.